application/common/models/User.php

```diff
@@ -0,0 +1,190 @@
+<?php
+
+namespace common\models;
+
+use common\helpers\Utils;
+use Exception;
+use yii\behaviors\AttributeBehavior;
+use yii\db\ActiveRecord;
+use yii\helpers\ArrayHelper;
+
+class User extends UserBase
+{
+    const SCENARIO_SAVE            = 'save';
+    const SCENARIO_UPDATE_PASSWORD = 'update_password';
+    const SCENARIO_AUTHENTICATE    = 'authenticate';
+
+    public $password;
+
+    public function scenarios(): array
+    {
+        $scenarios = parent::scenarios();
+
+        $scenarios[self::SCENARIO_DEFAULT] = [
+            'employee_id',
+            'first_name',
+            'last_name',
+            'display_name',
+            'username',
+            'email',
+            'active',
+            'locked',
+        ];
+
+        $scenarios[self::SCENARIO_UPDATE_PASSWORD] = ['password'];
+
+        $scenarios[self::SCENARIO_AUTHENTICATE] = ['username', 'password'];
+
+        return $scenarios;
+    }
+
+    public function rules(): array
+    {
+        return ArrayHelper::merge([
+            [
+                'active', 'default', 'value' => 'yes',
+            ],
+            [
+                'locked', 'default', 'value' => 'no',
+            ],
+            [
+                ['active', 'locked'], 'in', 'range' => ['yes', 'no'],
+            ],
+            [
+                'email', 'email',
+            ],
+            [
+                'password', 'required',
+                'on' => [self::SCENARIO_UPDATE_PASSWORD, self::SCENARIO_AUTHENTICATE],
+            ],
+            [
+                'password', 'string',
+            ],
+            [
+                'password',
+                $this->validatePassword(),
+                'on' => self::SCENARIO_AUTHENTICATE,
+            ],
+            [
+                ['last_synced_utc', 'last_changed_utc'],
+                'default', 'value' => Utils::now(),
+            ],
+            [
+                ['employee_id', 'first_name', 'last_name', 'display_name', 'username', 'email'],
```

Comment:
It looks like the parent::rules() already include a rule that these fields (plus `password_hash`) have to be strings. Was there still a need to include that rule here, too?